libs/core/src/lib/action-sheet/action-sheet.component.ts

```diff
@@ -0,0 +1,214 @@
+    @Output()
+    focusEscapeList = new EventEmitter<FocusEscapeDirection>();
+
+    /** Event thrown, when the action sheet is opened or closed */
+    @Output()
+    readonly openChange: EventEmitter<boolean> = new EventEmitter<boolean>();
+
+    /** @hidden */
+    @ContentChild(ActionSheetBodyComponent)
+    actionSheetBody: ActionSheetBodyComponent;
+
+    /** @hidden */
+    @ContentChild(ActionSheetControlComponent)
+    actionSheetControl: ActionSheetControlComponent;
+
+    /** @hidden */
+    @ContentChildren(ActionSheetItemComponent, { descendants: true })
+    actionSheetItems: QueryList<ActionSheetItemComponent>;
+
+    /** @hidden */
+    @ViewChild('actionSheetBodyTemplate')
+    actionSheetBodyTemplate: TemplateRef<any>;
+
+    /** @hidden */
+    @ViewChild(PopoverComponent)
+    popoverComponent: PopoverComponent;
+
+    /** @hidden */
+    actionSheetMobileDynamic: ComponentRef<ActionSheetMobileComponent>;
+
+    /** @hidden **/
+    private readonly _onDestroy$: Subject<void> = new Subject<void>();
+
+    /** @hidden **/
+    private readonly _onRefresh$: Subject<void> = new Subject<void>();
+
+    constructor(
+        private _elementRef: ElementRef,
+        private _keyboardSupportService: KeyboardSupportService<ActionSheetItemComponent>,
+        @Optional() private _dynamicComponentService: DynamicComponentService
+    ) {}
+
+
+    /** @hidden */
+    ngAfterContentInit(): void {
+        this._initializeChildrenState();
+        this._keyboardSupportService.setKeyboardService(this.actionSheetItems, false);
+        this._listenOnItemsChange();
+        this._actionControlHandle();
+        this._actionItemsHandle();
+    }
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        if (this.mobile) {
+            this._setUpMobileMode();
+        }
+        if (this.popoverComponent && this.popoverComponent) {
+            this.popoverComponent.directiveRef.loaded.pipe(takeUntil(this._onDestroy$)).subscribe(() => {
+                setTimeout(() => {
+                    this.setItemActive(0);
+                });
+            });
+        }
+    }
+
+    /** @hidden */
+    ngOnDestroy(): void {
+        this._onDestroy$.next();
+        this._onDestroy$.complete();
+        this.actionSheetControl.clicked.unsubscribe();
+        // this.actionSheetItems.unsubscribe();
+    }
+
+    /** @hidden */
+    private _initializeChildrenState(): void {
+        this.actionSheetBody.mobile = this.mobile;
+        this.actionSheetBody.compact = this.compact;
+        this.actionSheetItems.forEach(actionSheetItem => actionSheetItem.compact = this.compact);
+    }
+
+    /** @hidden */
+    private _actionControlHandle(): void {
+        this.actionSheetControl.clicked.subscribe((isOpen) => this.isOpenChangeHandle(isOpen));
+    }
+
+    /** @hidden */
+    private _actionItemsHandle(): void {
+        this.actionSheetItems.forEach(item => item.clicked.subscribe((isOpen) => this.isOpenChangeHandle(isOpen))
```

Comment:
Can you unsubscribe after destroy there? You can do it by adding `.pipe(takeUntil(this._onDestroy$))` like in other places in code